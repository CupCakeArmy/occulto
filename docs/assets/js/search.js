var typedoc = typedoc || {};
            typedoc.search = typedoc.search || {};
            typedoc.search.data = {"kinds":{"1":"External module","4":"Enumeration","16":"Enumeration member","32":"Variable","64":"Function","128":"Class","1024":"Property","2048":"Method","65536":"Type literal","4194304":"Type alias"},"rows":[{"id":0,"kind":1,"name":"\"Hash\"","url":"modules/_hash_.html","classes":"tsd-kind-external-module"},{"id":1,"kind":4,"name":"Hashes","url":"enums/_hash_.hashes.html","classes":"tsd-kind-enum tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"Hash\""},{"id":2,"kind":16,"name":"MD5","url":"enums/_hash_.hashes.html#md5","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":3,"kind":16,"name":"SHA1_1","url":"enums/_hash_.hashes.html#sha1_1","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":4,"kind":16,"name":"SHA1_256","url":"enums/_hash_.hashes.html#sha1_256","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":5,"kind":16,"name":"SHA1_512","url":"enums/_hash_.hashes.html#sha1_512","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":6,"kind":16,"name":"SHA3_256","url":"enums/_hash_.hashes.html#sha3_256","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":7,"kind":16,"name":"SHA3_384","url":"enums/_hash_.hashes.html#sha3_384","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":8,"kind":16,"name":"SHA3_512","url":"enums/_hash_.hashes.html#sha3_512","classes":"tsd-kind-enum-member tsd-parent-kind-enum tsd-is-not-exported","parent":"\"Hash\".Hashes"},{"id":9,"kind":128,"name":"Hash","url":"classes/_hash_.hash.html","classes":"tsd-kind-class tsd-parent-kind-external-module","parent":"\"Hash\""},{"id":10,"kind":1024,"name":"Hashes","url":"classes/_hash_.hash.html#hashes","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-static","parent":"\"Hash\".Hash"},{"id":11,"kind":2048,"name":"digest","url":"classes/_hash_.hash.html#digest","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"Hash\".Hash"},{"id":12,"kind":1,"name":"\"RSA\"","url":"modules/_rsa_.html","classes":"tsd-kind-external-module"},{"id":13,"kind":128,"name":"RSA","url":"classes/_rsa_.rsa.html","classes":"tsd-kind-class tsd-parent-kind-external-module","parent":"\"RSA\""},{"id":14,"kind":2048,"name":"gen","url":"classes/_rsa_.rsa.html#gen","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"RSA\".RSA"},{"id":15,"kind":2048,"name":"encrypt","url":"classes/_rsa_.rsa.html#encrypt","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"RSA\".RSA"},{"id":16,"kind":2048,"name":"decrypt","url":"classes/_rsa_.rsa.html#decrypt","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"RSA\".RSA"},{"id":17,"kind":4194304,"name":"PrivateKey","url":"modules/_rsa_.html#privatekey","classes":"tsd-kind-type-alias tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"RSA\""},{"id":18,"kind":4194304,"name":"PublicKey","url":"modules/_rsa_.html#publickey","classes":"tsd-kind-type-alias tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"RSA\""},{"id":19,"kind":4194304,"name":"KeyPair","url":"modules/_rsa_.html#keypair","classes":"tsd-kind-type-alias tsd-parent-kind-external-module","parent":"\"RSA\""},{"id":20,"kind":65536,"name":"__type","url":"modules/_rsa_.html#keypair.__type","classes":"tsd-kind-type-literal tsd-parent-kind-type-alias tsd-is-not-exported","parent":"\"RSA\".KeyPair"},{"id":21,"kind":32,"name":"pub","url":"modules/_rsa_.html#keypair.__type.pub","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"RSA\".KeyPair.__type"},{"id":22,"kind":32,"name":"prv","url":"modules/_rsa_.html#keypair.__type.prv","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"RSA\".KeyPair.__type"},{"id":23,"kind":1,"name":"\"Util\"","url":"modules/_util_.html","classes":"tsd-kind-external-module"},{"id":24,"kind":128,"name":"Base64","url":"classes/_util_.base64.html","classes":"tsd-kind-class tsd-parent-kind-external-module","parent":"\"Util\""},{"id":25,"kind":2048,"name":"encode","url":"classes/_util_.base64.html#encode","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"Util\".Base64"},{"id":26,"kind":2048,"name":"decode","url":"classes/_util_.base64.html#decode","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"Util\".Base64"},{"id":27,"kind":64,"name":"Rand","url":"modules/_util_.html#rand","classes":"tsd-kind-function tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"Util\""},{"id":28,"kind":1,"name":"\"Symmetric\"","url":"modules/_symmetric_.html","classes":"tsd-kind-external-module"},{"id":29,"kind":4,"name":"Ciphers","url":"enums/_symmetric_.ciphers.html","classes":"tsd-kind-enum tsd-parent-kind-external-module","parent":"\"Symmetric\""},{"id":30,"kind":16,"name":"ChaCha20","url":"enums/_symmetric_.ciphers.html#chacha20","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":31,"kind":16,"name":"AES_256_GCM","url":"enums/_symmetric_.ciphers.html#aes_256_gcm","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":32,"kind":16,"name":"AES_192_GCM","url":"enums/_symmetric_.ciphers.html#aes_192_gcm","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":33,"kind":16,"name":"AES_128_GCM","url":"enums/_symmetric_.ciphers.html#aes_128_gcm","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":34,"kind":16,"name":"AES_256_CTR","url":"enums/_symmetric_.ciphers.html#aes_256_ctr","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":35,"kind":16,"name":"AES_192_CTR","url":"enums/_symmetric_.ciphers.html#aes_192_ctr","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":36,"kind":16,"name":"AES_128_CTR","url":"enums/_symmetric_.ciphers.html#aes_128_ctr","classes":"tsd-kind-enum-member tsd-parent-kind-enum","parent":"\"Symmetric\".Ciphers"},{"id":37,"kind":128,"name":"Symmetric","url":"classes/_symmetric_.symmetric.html","classes":"tsd-kind-class tsd-parent-kind-external-module","parent":"\"Symmetric\""},{"id":38,"kind":1024,"name":"Ciphers","url":"classes/_symmetric_.symmetric.html#ciphers","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-static","parent":"\"Symmetric\".Symmetric"},{"id":39,"kind":1024,"name":"Encoding","url":"classes/_symmetric_.symmetric.html#encoding","classes":"tsd-kind-property tsd-parent-kind-class tsd-is-static","parent":"\"Symmetric\".Symmetric"},{"id":40,"kind":2048,"name":"encrypt","url":"classes/_symmetric_.symmetric.html#encrypt","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"Symmetric\".Symmetric"},{"id":41,"kind":2048,"name":"decrypt","url":"classes/_symmetric_.symmetric.html#decrypt","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-static","parent":"\"Symmetric\".Symmetric"},{"id":42,"kind":2048,"name":"getCipherConfig","url":"classes/_symmetric_.symmetric.html#getcipherconfig","classes":"tsd-kind-method tsd-parent-kind-class tsd-is-private tsd-is-static","parent":"\"Symmetric\".Symmetric"},{"id":43,"kind":4194304,"name":"CipherConfig","url":"modules/_symmetric_.html#cipherconfig","classes":"tsd-kind-type-alias tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"Symmetric\""},{"id":44,"kind":65536,"name":"__type","url":"modules/_symmetric_.html#cipherconfig.__type","classes":"tsd-kind-type-literal tsd-parent-kind-type-alias tsd-is-not-exported","parent":"\"Symmetric\".CipherConfig"},{"id":45,"kind":32,"name":"alg","url":"modules/_symmetric_.html#cipherconfig.__type.alg","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".CipherConfig.__type"},{"id":46,"kind":32,"name":"keySize","url":"modules/_symmetric_.html#cipherconfig.__type.keysize","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".CipherConfig.__type"},{"id":47,"kind":32,"name":"ivSize","url":"modules/_symmetric_.html#cipherconfig.__type.ivsize","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".CipherConfig.__type"},{"id":48,"kind":32,"name":"mac","url":"modules/_symmetric_.html#cipherconfig.__type.mac","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".CipherConfig.__type"},{"id":49,"kind":4194304,"name":"EncryptedItem","url":"modules/_symmetric_.html#encrypteditem","classes":"tsd-kind-type-alias tsd-parent-kind-external-module tsd-is-not-exported","parent":"\"Symmetric\""},{"id":50,"kind":65536,"name":"__type","url":"modules/_symmetric_.html#encrypteditem.__type-1","classes":"tsd-kind-type-literal tsd-parent-kind-type-alias tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem"},{"id":51,"kind":32,"name":"alg","url":"modules/_symmetric_.html#encrypteditem.__type-1.alg-1","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":52,"kind":32,"name":"data","url":"modules/_symmetric_.html#encrypteditem.__type-1.data","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":53,"kind":32,"name":"iv","url":"modules/_symmetric_.html#encrypteditem.__type-1.iv","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":54,"kind":32,"name":"salt","url":"modules/_symmetric_.html#encrypteditem.__type-1.salt","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":55,"kind":32,"name":"keySize","url":"modules/_symmetric_.html#encrypteditem.__type-1.keysize-1","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":56,"kind":32,"name":"tag","url":"modules/_symmetric_.html#encrypteditem.__type-1.tag","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":57,"kind":32,"name":"tagSize","url":"modules/_symmetric_.html#encrypteditem.__type-1.tagsize","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"Symmetric\".EncryptedItem.__type"},{"id":58,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-external-module"},{"id":59,"kind":32,"name":"RSA","url":"modules/_index_.html#rsa","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"index\""},{"id":60,"kind":32,"name":"Symmetric","url":"modules/_index_.html#symmetric","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"index\""},{"id":61,"kind":32,"name":"Hash","url":"modules/_index_.html#hash","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"index\""},{"id":62,"kind":32,"name":"Util","url":"modules/_index_.html#util","classes":"tsd-kind-variable tsd-parent-kind-external-module","parent":"\"index\""},{"id":63,"kind":65536,"name":"__type","url":"modules/_index_.html#util.__type","classes":"tsd-kind-type-literal tsd-parent-kind-variable tsd-is-not-exported","parent":"\"index\".Util"},{"id":64,"kind":32,"name":"Rand","url":"modules/_index_.html#util.__type.rand","classes":"tsd-kind-variable tsd-parent-kind-type-literal tsd-is-not-exported","parent":"\"index\".Util.__type"}]};