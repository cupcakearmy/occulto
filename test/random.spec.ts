import { describe, expect, it } from 'vitest'
import { getRandomBytes } from '../dist/index.js'

describe('Random', () => {
  it('Should be able to create random values', async () => {
    const buffer = await getRandomBytes(8)
    expect(buffer).instanceOf(Uint8Array)
    expect(buffer.byteLength).toEqual(8)
  })

  it('Should throw error on empty array', async () => {
    await expect(() => getRandomBytes(0)).rejects.toThrowErrorMatchingSnapshot()
  })

  it('Should throw error on negative bytes', async () => {
    await expect(() => getRandomBytes(-1)).rejects.toThrowErrorMatchingSnapshot()
  })
})
