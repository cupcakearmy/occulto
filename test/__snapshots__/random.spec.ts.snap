// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Random > Should throw error on empty array 1`] = `[Error: Invalid number of bytes]`;

exports[`Random > Should throw error on negative bytes 1`] = `[Error: Invalid number of bytes]`;
